rpc/client.go

```diff
@@ -19,31 +19,48 @@ type rpcClient struct {
 	clientCloser jasper.CloseFunc
 }
 
-// NewClient creates a connection to the RPC service specified
-// in the address. If certFile is non-empty, the credentials will be read from
-// the file to establish a secure TLS connection; otherwise, it will establish
-// an insecure connection. The caller is responsible for closing the connection
-// using the returned jasper.CloseFunc.
-func NewClient(ctx context.Context, addr net.Addr, certFile string) (jasper.RemoteClient, error) {
-	var credsDialOpt grpc.DialOption
-	if certFile != "" {
-		creds, err := credentials.NewClientTLSFromFile(certFile, "")
+// NewClient creates a connection to the RPC service with the specified address
+// addr. If creds is non-nil, the credentials will be used to establish a secure
+// TLS connection with the service; otherwise, it will establish an insecure
+// connection. The caller is responsible for closing the connection using the
+// returned jasper.CloseFunc.
+func NewClient(ctx context.Context, addr net.Addr, creds *Credentials) (jasper.RemoteClient, error) {
+	opts := []grpc.DialOption{grpc.WithBlock()}
+	if creds != nil {
+		tlsConf, err := creds.Resolve()
 		if err != nil {
-			return nil, errors.Wrapf(err, "could not get client credentials from cert file '%s'", certFile)
+			return nil, errors.Wrap(err, "could not resolve credentials into TLS config")
 		}
-		credsDialOpt = grpc.WithTransportCredentials(creds)
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(tlsConf)))
```

Comment:
Retry interceptors?

Reply:
Aviation retry logic only retires on [these errors](https://github.com/evergreen-ci/aviation/blob/master/retry.go#L120-L134). Would it still be too risky to retry? If the grpc request to spawn a host fails, can the host still have been spawned?

Reply:
I'm not going to do this for now because a lot of the RPCs will return `codes.Unknown` and aviation will retry on these errors when it shouldn't (and this will slow down tests a lot with unnecessary retrying). I made a [separate ticket](https://jira.mongodb.org/browse/MAKE-814) to handle request retries and fix up the RPCs to work with retrying more nicely.